### ISSUE TITLE
supplying my own fetch implementation doesn't work at node

### ISSUE THREAD

**Issue by haoliangyu** [Esri/arcgis-rest-js#203]
The documentation does not say I cannot provide my own `fetch` implementation at node. So I try

``` javascript
require("isomorphic-form-data");

const fetch = require('node-fetch')
const { request } = require("@esri/arcgis-rest-request");

request("https://www.arcgis.com/sharing/rest/info", { fetch }).then(response => {
  console.log(response);
});
```

then it throws an error

```
/home/linux/Desktop/Untitled Folder/node_modules/@esri/arcgis-rest-request/dist/node/request.js:49
    var options = tslib_1.__assign({ httpMethod: "POST", fetch: fetch }, requestOptions);
                                                                ^

ReferenceError: fetch is not defined
    at request (/home/linux/test/node_modules/@esri/arcgis-rest-request/dist/node/request.js:49:65)
    at Object.<anonymous> (/home/linux/test/index.js:7:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3
```

My node version is v8.11.1 and it does not seem to like the undeclared variable `fetch`.

**Comment by jgravois** [comment]
i can repro this. the only way i could use this lib w/ `node-fetch` successfully was via the global: 

```js
"use strict";
/* Copyright (c) 2017 Environmental Systems Research Institute, Inc.
 * Apache-2.0 */
Object.defineProperty(exports, "__esModule", { value: true });
// ...
var fetch = require('node-fetch');
var Headers = fetch.Headers;  // this isn't necessary with `isomorphic-fetch`
```

**Comment by jgravois** [comment]
addressed (at least partially) by #263

**Comment by jgravois** [comment]
addressed (fully) by #264
